bootstrap.c

```diff
@@ -81,25 +82,86 @@ static const char *get_host_triple(void) {
     return global_buffer;
 }
 
+#if defined(__WIN32__)
+#define HOST_PATHSEP '\\'
+#else
+#define HOST_PATHSEP '/'
+#endif
+
+static const char *get_parent_dir(const char *cwd) {
+    // safe for our use case
+    static char global_buffer[255];
+    if (strlen(cwd) + 3 >= 255)
+        panic("unable to get parent directory");
+
+    sprintf(global_buffer, "%s%c..", cwd, HOST_PATHSEP);
+    return global_buffer;
+}
+
+static const char *join(const char *dirpath, const char *path) {
+    int size = strlen(dirpath) + strlen(path) + 1;
+    if (size >= 4096)
+        panic("unable to join dirpath and path");
+
+    char *buffer = malloc(size);
+    sprintf(buffer, "%s%c%s", dirpath, HOST_PATHSEP, path);
+
+    // memory is leaked
+    return buffer;
+}
+
+static const char *find_zig_root(void) {
+    const char *cwd = ".";
+    while (true) {
+        cwd = get_parent_dir(cwd);
+        FILE *f = fopen(join(cwd, "stage1/zig1.wasm"), "r");
+        if (f != NULL) {
+            fclose(f);
+            return cwd;
+        }
+    }
+
+    // unreachable
```

Comment:
sounds like this should be panic

Reply:
It was a bug: the code never checked if zig is in the current directory.